virtual_rainforest/models/litter/litter_model.py

```diff
@@ -106,7 +107,8 @@ def __init__(
         update_interval: Quantity,
         soil_layers: list[float],
         canopy_layers: int,
-        constants: LitterConsts,
+        model_constants: LitterConsts,
+        core_constants: CoreConsts,
```

Comment:
Ahh yeah when I was writing it I did find having to track both to be pretty clumsy. So, a different approach definitely makes sense. I guess that the difficultly lies in setting it up in such a way so that the value of constants can still be changed using the configuration.

I guess one option would be for the constants classes to take a configured `CoreConsts` object as an argument, e.g. 

```python
class LitterConsts(core_constants: CoreConsts)
    ...
    depth_of_active_soil_layer: float = core_constants.depth_of_active_soil_layer
```

Then the configuration process could configure `CoreConsts` first (which could change the value of `depth_of_active_soil_layer`), and use this to generate model specific constants data classes.

Reply:
The problem with that is that `LitterConst` is a `dataclass` so its inputs are already the constant values it contains. however, it might be a step in the right direction if combined with `properties`:

```python
class LitterConsts:
    ...
    core_constants: CoreConsts = CoreConsts()

    @property
    def depth_of_active_soil_layer(self) -> float:
        return core_constants.depth_of_active_soil_layer
```

So `core_constants` becomes just another attribute of the `LitterConst` dataclass, with a default value of `CoreConst()` but that can get as input a modified `CoreConst` object, if present. Then, the constants of interest are accessed with properties.

Reply:
Okay makes sense, should I make an issue for this then?

Reply:
I've backed away from this slightly - see commments on #342. I think having a `coreConsts` as an input to `BaseModel` is much more direct than expecting each model to merge in core constants as properties of a model specific class. Both work, though, so I don't think there's too much to worry about here.